wsl_cheatsheet.md

```diff
@@ -18,6 +19,43 @@ The goal is to gather here all tips and known issues about WSL2!
 
 # Issues
 
+## Network Unreachable
+
+When trying to connect to remote hosts, eg. when trying to do `sudo apt update`, you might see an error message saying that the network is unreachable. Here are two possible solutions: if the first one doesn't help, try the second one.
+
+### A note about diagnosing this problem
+- This is a common networking issue, and we must be sure that the problem lies with WSL only, and not with the Windows host. Please first ensure that there are no known networking issues on Windows. If you can successfully browse a few sites on Windows, but you can't connect to the internet in WSL, it would appear that this is an WSL problem. Please continue to the next section.
+
+### Refresh and reset network
+- `Start` > `cmd` > 'Open as administrator'
+- Run the following commands:
+    ```bash
+    wsl --shutdown
+    netsh winsock reset
+    netsh int ip reset all
+    netsh winhttp reset proxy
+    ipconfig /flushdns
+    ```
+- Reset the network (`Start` > Search for `Network Reset`; click `Reset`)
+- Restart Windows
+- *[Source](https://github.com/microsoft/WSL/issues/4926#issuecomment-679410653)*. There might be other relevant solutions there.
+- If this still does not solve your problem, try the next section.
+
+### Add a custom DNS server address to WSL
+- Open a WSL terminal and type the following:
+  ```bash
+  sudo mv /etc/resolv.conf /etc/resolv.conf.old
+  sudo touch /etc/resolv.conf
+  sudo echo "[network]\ngenerateResolvConf = false\n" >> /etc/resolv.conf
```

Comment:
Hello 👋

Thank you for contributing.

I could be wrong but I think that `/etc/resolv.conf` is automatically generated by WSL. Anything modified in there would be overridden on the next restart.

The real fix here is the `wsl --shutdown`

Can we remove the second part about adding a custom DNS server? 🙏 

Many thanks.

Reply:
Hi there :)
You are correct; `/etc/resolv.conf` is auto-generated by WSL. This can be prevented by line 49 above - however, I realize that the end of that line should read `/etc/wsl.conf`, not `/etc/resolv.conf`. I'll change that and update.

Reply:
(See [this solution](https://github.com/microsoft/WSL/issues/5420#issuecomment-646479747), which describes the same thing I am suggesting in this PR)

Reply:
Just to clarify - the second part does not add a custom DNS server; it adds the address of a custom DNS server to be used for resolving host names by WSL. For whatever reason, the auto-generated one does not work, and this prevents internet access by WSL.

Also, `wsl --shutdown` did not solve the problem without doing the steps I outlined in the second part ("Add a custom DNS server address to WSL")